triton.go

```diff
@@ -0,0 +1,18 @@
+package triton
+
+import (
+	"github.com/joyent/triton-go/authentication"
+)
+
+// Universal package used for defining configuration used across all client
+// constructors.
+
+// ClientConfig is a placeholder/input struct around the behavior of configuring
+// a client constructor through the implementation's runtime environment
+// (SDC/MANTA env vars).
+type ClientConfig struct {
+	Endpoint    string
```

Comment:
Should we rename this to `TritonURL` to mirror `MantaURL`? Since this would be breaking, now would be the best time to do this.